packages/compiler-cli/src/ngtsc/annotations/src/component.ts

```diff
@@ -361,6 +361,14 @@ export class ComponentDecoratorHandler implements
     // Register this component's information with the `MetadataRegistry`. This ensures that
     // the information about the component is available during the compile() phase.
     const ref = new Reference(node);
+
+    let animations: Map<string, string>[]|undefined;
```

Comment:
I'd propose to use `null` instead of `undefined` here:
```suggestion
    let animations: Map<string, string>[]|null = null;
```
and also make the `animations` field of the `DirectiveMeta` structure non-optional. This should be more inline with the rest of the code.